server/src/test/java/org/opensearch/index/store/remote/utils/TransferManagerTestCase.java

```diff
@@ -64,6 +65,25 @@ protected static byte[] createData() {
         return data;
     }
 
+    public void testOverflowDisabled() throws Exception {
+        FileCache strictCache = FileCacheFactory.createConcurrentLRUFileCache(
+            EIGHT_MB * 2,
+            1,
+            new NoopCircuitBreaker(CircuitBreaker.REQUEST)
+        );
+        strictCache.enableOverflow(false);
+        TransferManager tm = initializeTransferManager(strictCache);
+
+        assertThrows(IOException.class, () -> {
+            IndexInput i1 = fetchBlobWithName(tm, "1");
+            IndexInput i2 = fetchBlobWithName(tm, "2");
+            IndexInput i3 = fetchBlobWithName(tm, "3");
+            assertIndexInputIsFunctional(i1);
+            assertIndexInputIsFunctional(i2);
+            assertIndexInputIsFunctional(i3);
```

Comment:
Anyone of these lines could throw an IOException and this test would pass. You should scope the runnable passed to `assertThrows()` to be as small as possible so you're asserting exactly what you expect to throw does in fact throw. In this case it's unclear which of `i1`, `i1`, or `i3` is expected to fail.